hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -4815,8 +4815,11 @@ public int determineTableNumberForColumn(String columnName) {
 
 	@Override
 	public EntityEntryFactory getEntityEntryFactory() {
```

Comment:
Here again... if this is such performance critical code, why even do the if check?  We know this already when the EntityPersister is constructed.  So just do that if check there and store the EntityEntryFactory as a inst var.

Also we discussed quite a few times now... the terms here should be mutable and immutable.  Not default and immutable.